### ISSUE TITLE
[clang-tidy][false-positive] `misc-const-correctness` suggests converting return value to pointer-to-const with return type `void*`

### ISSUE THREAD

**Issue by tearfur** [llvm/llvm-project#206429]
When a pointer variable is returned in a function that has return type `void*`, `misc-const-correctness` suggests to convert the pointer variable into pointer-to-const, but this is a syntax error.

https://godbolt.org/z/MWTKbGx8n

```
<source>:2:5: warning: pointee of variable 'ip' of type 'int *' can be declared 'const' [misc-const-correctness]
    2 |     int* ip = nullptr;
      |     ^  
      |         const
1 warning generated.
```

Current trunk: 41ef9d77f7826ec45e83ab4c6b83a96d735830c6
Looks a lot like a case that was missed in #170065.
